package/MDAnalysis/analysis/atomicdistances.py

```diff
@@ -163,14 +168,21 @@ def __init__(self, ag1, ag2, pbc=True, **kwargs):
         self._ag1 = ag1
         self._ag2 = ag2
         self._pbc = pbc
+        self.results = Results()
 
     def _prepare(self):
         # initialize NumPy array of frames x distances for results
-        self.results = np.zeros((self.n_frames, self._ag1.atoms.n_atoms))
+        self.results.distances = np.zeros((
+            self.n_frames, self._ag1.atoms.n_atoms
+        ))
 
     def _single_frame(self):
         # if PBCs considered, get box size
         box = self._ag1.dimensions if self._pbc else None
-        self.results[self._frame_index] = calc_bonds(self._ag1.positions,
-                                                     self._ag2.positions,
-                                                     box)
+        self.results.distances[self._frame_index] = calc_bonds(
+            self._ag1.positions, self._ag2.positions, box
+        )
+
+    def _conclude(self):
+        # adjust self.results to self.results.distances
+        self.results = self.results.distances
```

Comment:
Maybe I'm just confused but it looks to me like the tests haven't actually changed yet (just formatting changes as far as I can tell?) and this just restored the original behavior of `self.results` storing a NumPy array again?

Reply:
yes, first I modified it to have the output would be `self.results.distances`, but then I thought that for the convenience to not modify the docs, where it uses `my_dists.results` to `my_dists.results.distances` I could define `self.results` as `self.results.distances` to not have to change the docs and still be able to implement the parallelization of the class.

As for the `pytest` changes, first I had the `self.results.distances` output and thus the tests needed to have also the `.distances`. The Idea to have `self.results` came after creating the PR :see_no_evil:, but yes I will then remove the mention of the `pytest` modification, was quite late when I finished up the PR, so didn't think about that :)